packages/run-protocol/src/runStake/attestation.js

```diff
@@ -0,0 +1,254 @@
+    );
+    fit(attestationAmount.value.payload, harden([[M.string(), M.bigint()]]));
+    /** @type {[string, bigint][]} */
+    const [[address, valueReturned]] = attestationAmount.value.payload;
+    const lienedAmount = AmountMath.make(stakeBrand, valueReturned);
+    return { address, lienedAmount };
+  };
+
+  /** @type {(m: Amount, s: Amount) => Amount} */
+  const subtractOrZero = (m, s) => (isGTE(m, s) ? subtract(m, s) : empty);
+  // check that x <= hi - lo
+  const checkDelta = (x, label, hi, lo) => {
+    const bound = subtractOrZero(hi, lo);
+    assert(
+      isGTE(bound, coerce(stakeBrand, x)),
+      // TODO: X`...${raw(label)}...`
+      X`Only ${bound} was ${label}, but an attestation was attempted for ${x}`,
+    );
+  };
+  const assertAccountStateOK = async (address, toLien) => {
+    const s = await E(lienBridge).getAccountState(address, stakeBrand);
+    const unlocked = subtractOrZero(s.total, s.locked);
+    checkDelta(toLien, 'unliened', s.total, s.liened);
+    checkDelta(toLien, 'bonded and unliened', s.bonded, s.liened);
+    checkDelta(toLien, 'unlocked and unliened', unlocked, s.liened);
+  };
+
+  /** @type {Store<Address, Amount<'nat'>>} */
+  const amountByAddress = makeStore('amount');
+
+  /**
+   * @param { string } address
+   * @param { Amount<'nat'> } lienDelta
+   */
+  const mintAttestation = async (address, lienDelta) => {
+    assert.typeof(address, 'string');
+    // This account state check is primarily to provide useful diagnostics.
+    // Since things like the bonded amount can change out from under us,
+    // we shouldn't rely on it completely.
+    // TODO: consider performance implications vs. diagnostics.
+    await assertAccountStateOK(address, lienDelta);
+
+    const current = getOrElse(amountByAddress, address, () => empty);
+    const target = add(current, lienDelta);
+
+    // Since our mint is not subject to normal supply/demand,
+    // minting here and failing in setLiened below does no harm.
+    const attestationPaymentP = mintZCFMintPayment(
+      zcf,
+      zcfMint,
+      wrapLienedAmount(address, lienDelta),
+    );
+
+    // If setLiened returns successfully, then
+    // the preconditions were indeed met and we
+    // can update the stored liened amount.
+    await E(lienBridge).setLiened(address, current, target);
+    amountByAddress.set(address, target);
+    return attestationPaymentP;
+  };
+
+  /** @param {ZCFSeat} seat */
+  const returnAttestation = async seat => {
+    assertProposalShape(seat, { give: { Attestation: null }, want: {} });
+
+    /** @type {Amount<'copyBag'>} */
+    const attestationAmount = seat.getAmountAllocated(KW.Attestation);
+    const { address, lienedAmount: amountReturned } =
+      unwrapLienedAmount(attestationAmount);
+
+    const liened = amountByAddress.get(address); // or throw
+    const updated = subtract(liened, amountReturned); // or throw
```

Comment:
took me a tick to realize "updated what" and that these were pre and post values. consider,
```suggestion
    const lienedPre = amountByAddress.get(address); // or throw
    const lienedPost = subtract(lienedPre, amountReturned); // or throw
```